packages/liveblocks-react/src/room.tsx

```diff
@@ -1085,31 +1085,30 @@ export function createRoomContext<
       roomId: room.id,
     });
 
-    store.set((state) => ({
-      ...state,
-      threadsQueries: {
-        ...state.threadsQueries,
-        [queryKey]: {
-          isLoading: true,
-        },
-      },
-    }));
-
-    // TODO: Error handling
-    const { threads, inboxNotifications } = await initialPromise;
+    store.setThreadsQueryState(queryKey, {
+      isLoading: true,
+    });
 
-    store.updateThreadsAndNotifications(
-      Object.fromEntries(threads.map((thread) => [thread.id, thread])),
-      Object.fromEntries(
-        inboxNotifications.map((notification) => [
-          notification.id,
-          notification,
-        ])
-      ),
-      queryKey
-    );
+    try {
+      const { threads, inboxNotifications } = await initialPromise;
+      store.updateThreadsAndNotifications(
+        Object.fromEntries(threads.map((thread) => [thread.id, thread])),
+        Object.fromEntries(
+          inboxNotifications.map((notification) => [
+            notification.id,
+            notification,
+          ])
+        ),
+        queryKey
+      );
 
-    poller.start(POLLING_INTERVAL);
+      poller.start(POLLING_INTERVAL);
+    } catch (err) {
+      store.setThreadsQueryState(queryKey, {
+        isLoading: false,
+        error: err as Error,
```

Comment:
**TODO**: If the initial request failed but polling was successful, we want to clear this `error` variable.

Reply:
Done: https://github.com/liveblocks/liveblocks/pull/1375/commits/7cbfa0ea5ee994ead736f709e21e84190eac875f